sections/header.liquid

```diff
@@ -345,25 +345,25 @@
             <li>
               {%- if link.links != blank -%}
                 <details-disclosure>
-                  <details>
+                  <details id="Details-HeaderMenu">
                     <summary class="header__menu-item list-menu__item link focus-inset">
                       <span {%- if link.child_active %} class="header__active-menu-item"{% endif %}>{{ link.title | escape }}</span>
                       {% render 'icon-caret' %}
                     </summary>
-                    <ul class="header__submenu list-menu list-menu--disclosure caption-large motion-reduce" role="list" tabindex="-1">
+                    <ul id="HeaderMenu-MenuList" class="header__submenu list-menu list-menu--disclosure caption-large motion-reduce" role="list" tabindex="-1">
                       {%- for childlink in link.links -%}
                         <li>
                           {%- if childlink.links == blank -%}
                             <a href="{{ childlink.url }}" class="header__menu-item list-menu__item link link--text focus-inset caption-large{% if childlink.current %} list-menu__item--active{% endif %}"{% if childlink.current %} aria-current="page"{% endif %}>
                               {{ childlink.title | escape }}
                             </a>
                           {%- else -%}
-                            <details>
+                            <details id="Details-HeaderSubMenu-{{ forloop.index }}">
                               <summary class="header__menu-item link link--text list-menu__item focus-inset caption-large">
                                 {{ childlink.title | escape }}
                                 {% render 'icon-caret' %}
                               </summary>
-                              <ul class="header__submenu list-menu motion-reduce">
+                              <ul id="HeaderMenu-SubMenuList-{{ forloop.index }}" class="header__submenu list-menu motion-reduce">
                                 {%- for grandchildlink in childlink.links -%}
```

Comment:
Should the same approach be used for the mobile menu ? And we can then remove the `addAccessibilityAttributes(summaryElements)` function from the `MenuDrawer` class.